### DISCUSSION TITLE
the ID of the first TimelineNode always is beginning of 0, can i change it?

### DISCUSSION THREAD

**Discussion by mupsycho** [jspsych/jsPsych#1487] [Q&A]
like this data
```json
[
	{
		"rt": 819,
		"stimulus": "0",
		"button_pressed": 0,
		"trial_type": "html-button-response",
		"trial_index": 0,
		"time_elapsed": 820,
		"internal_node_id": "0.0-0.0"
	},
	{
		"rt": 181,
		"stimulus": "1",
		"button_pressed": 0,
		"trial_type": "html-button-response",
		"trial_index": 1,
		"time_elapsed": 1000,
		"internal_node_id": "0.0-1.0-0.0"
	},
	{
		"rt": 201,
		"stimulus": "2",
		"button_pressed": 0,
		"trial_type": "html-button-response",
		"trial_index": 2,
		"time_elapsed": 1202,
		"internal_node_id": "0.0-2.0-0.0"
	},
	{
		"rt": 203,
		"stimulus": "3",
		"button_pressed": 0,
		"trial_type": "html-button-response",
		"trial_index": 3,
		"time_elapsed": 1404,
		"internal_node_id": "0.0-3.0"
	}
]
```
the inernal_node_id always starts with "0.0", does this mean that I cannot modify the number of the top-level timeline？
I noticed this code in `jspsych.js`
```javascript
    this.ID = function() {
      var id = "";
      if (typeof parent_node == 'undefined') {
        return "0." + progress.current_iteration;
      } else {
        id += parent_node.ID() + "-";
        id += relative_id + "." + progress.current_iteration;
        return id;
      }
    }
```

**Comment by jodeleeuw** [comment]
That's right. This isn't really meant to be a piece of data that gets used in analysis work. jsPsych currently uses it in order to handle some searches within the data structure. I'm hoping to change that in the future and then remove this. We may remove it from any data exports and just use it internally as an interim solution.

Are you using this for some kind of information that isn't available elsewhere?

**Reply by mupsycho** [reply]
When I use `jsPsych.addNodeToEndOfTimeline` to add a timeline, it can only be added to the end of the top-level timeline. Although I can meet my needs now, I think `addNodeToEndOfTimeline` can be added to the end of the upper-level timeline by default. It is more convenient to fulfill my needs. From the perspective of data analysis, it has no effect.

**Comment by jodeleeuw** [comment]
I'm converting this over to a discussion.
